ci/jenkinsfiles/common-lib.groovy

```diff
@@ -0,0 +1,241 @@
+/*
+* (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* Contributors:
+*     Abdoul BA <aba@nuxeo.com>
+*/
+
+void compile() {
+  echo '''
+    ----------------------------------------
+    Compile
+    ----------------------------------------
+  '''
+  sh "mvn ${MAVEN_ARGS} -V -T0.8C -DskipTests clean install"
+}
+
+String getVersion() {
+  return this.isPullRequest() ? this.getPullRequestVersion() : this.getReleaseVersion()
+}
+
+String getReleaseVersion() {
+  String preid = 'rc'
+  String nextPromotion = readMavenPom().getVersion().replace('-SNAPSHOT', '')
+  String version = "${nextPromotion}-${preid}.0" // first version ever
+
+  // find the latest tag if any
+  sh "git fetch origin 'refs/tags/v${nextPromotion}*:refs/tags/v${nextPromotion}*'"
+  String tag = sh(returnStdout: true, script: "git tag --sort=taggerdate --list 'v${nextPromotion}*' | tail -1 | tr -d '\n'")
+  if (tag) {
+    container('maven') {
+      version = sh(returnStdout: true, script: "npx --ignore-existing semver -i prerelease --preid ${preid} ${tag} | tr -d '\n'")
+    }
+  }
+  return version
+}
+
+String getPullRequestVersion() {
+  return "${readMavenPom().getVersion()}-${BRANCH_NAME}"
+}
+
+void gitCheckout(String reference) {
+  sh "git checkout ${reference}"
+}
+
+void gitCommit(String message, String option) {
+  echo '''
+    ----------------------------------------
+    Git Commit
+    ----------------------------------------
+  '''
+  sh """
+    git commit ${option} -m "${message}"
+  """
+}
+
+void gitPush(String reference) {
+  sh "git push origin ${reference}"
+}
+
+void gitTag(String tagname, String message) {
+  sh """
+    git tag -a ${tagname} -m "${message}"
+  """
+}
+
+def isPullRequest() {
+  return "${BRANCH_NAME}" =~ /PR-.*/
+}
+
+void lint() {
+  echo '''
+    ----------------------------------------
+    Run Linting Validations
+    ----------------------------------------
+  '''
+  sh """
+    cd ${FRONTEND_FOLDER}
+    npm run lint
+  """
+}
+
+String mavenArgs() {
+  String args = '-DskipITs=true -fae -B -nsu'
+  return args
+}
+
+boolean needsSaucelabs() {
+  return this.isPullRequest() && pullRequest.labels.contains('saucelabs')
+}
+
+void runFrontEndUnitTests() {
+  return {
+    stage('Run Unit tests: FrontEnd') {
+      container('maven') {
```

Comment:
Should we still use this container here?

Reply:
Yes, we do,

before running the unit tests, we need to retrieve the sauce labs credential in case of.
and jx command is only available when using a container (playwright doesn't not handle jx command). 
https://github.com/nuxeo/nuxeo-coldstorage/blob/improvement-NXP-29783-configure-ci/ci/jenkinsfiles/common-lib.groovy#L121